source/app/powertabeditor.cpp

```diff
@@ -1366,6 +1367,28 @@ void PowerTabEditor::editRepeatEnding()
     }
 }
 
+void PowerTabEditor::updateDynamic(Dynamic::VolumeLevel volume)
+{
+    ScoreLocation &location = getLocation();
+    const Dynamic *dynamic = ScoreUtils::findByPosition(
+                location.getStaff().getDynamics(), location.getPositionIndex());
+
+    if (dynamic)
+    {
+        myUndoManager->push(new EditDynamic(location, volume, dynamic),
+                            location.getSystemIndex());
+    }
+    else
+    {
+        if (volume == dynamic->getVolume())
+            myUndoManager->push(new RemoveDynamic(location),
+                                location.getSystemIndex());
+        else
+            myUndoManager->push(new AddDynamic(location, dynamic),
```

Comment:
Here you need to construct a `Dynamic` since the pointer `dynamic` is `nullptr` due to the check above.
It should look something like:
```
else
{
    myUndoManager->push(new AddDynamic(location, Dynamic(location.getPositionIndex(), volume)), location.getSystemIndex());
}
```